src/lib/core/bidi/directionality.spec.ts

```diff
@@ -0,0 +1,120 @@
+import {async, fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {Component, getDebugNode} from '@angular/core';
+import {Dir, Directionality} from './index';
+
+function initDir () {
+  document.documentElement.dir = '';
+  document.body.dir = '';
+}
+
+describe('Directionality', () => {
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      providers: [Directionality],
+      declarations: [TestAppWithoutDir]
+    });
+
+    TestBed.compileComponents();
+
+    initDir();
```

Comment:
There needs to be an `afterEach` that restores the `dir` attributes to whatever they were before.